### ISSUE TITLE
Error upscaling grayscale image since v1.1.5

### ISSUE THREAD

**Issue by Its-my-right** [the-database/MangaJaNaiConverterGui#33]
Since v1.1.5 I can't upscale grayscale images, I get the following error (everything was working fine with the same params in v1.1.3 and older versions):

Process Process-2:
Traceback (most recent call last):
  File "C:\Users\Unknown\AppData\Roaming\MangaJaNaiConverterGui\python\python\Lib\site-packages\multiprocess\process.py", line 314, in _bootstrap
    self.run()
  File "C:\Users\Unknown\AppData\Roaming\MangaJaNaiConverterGui\python\python\Lib\site-packages\multiprocess\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py", line 1167, in upscale_worker
    image = ai_upscale_image(image, model_tile_size, model)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py", line 420, in ai_upscale_image
    return upscale_image_node(
           ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\packages\chaiNNer_pytorch\pytorch\processing\upscale_image.py", line 280, in upscale_image_node
    return basic_upscale(
           ^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\upscale\basic_upscale.py", line 106, in basic_upscale
    img = _custom_scale_upscale(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\upscale\basic_upscale.py", line 57, in _custom_scale_upscale
    return upscale(img)
           ^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\upscale\basic_upscale.py", line 92, in inner_upscale
    return convenient_upscale(
           ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\upscale\convenient_upscale.py", line 98, in convenient_upscale
    return upscale(img)
           ^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\packages\chaiNNer_pytorch\pytorch\processing\upscale_image.py", line 282, in <lambda>
    lambda i: upscale(
              ^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\packages\chaiNNer_pytorch\pytorch\processing\upscale_image.py", line 100, in upscale
    img_out = pytorch_auto_split(
              ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Roaming\MangaJaNaiConverterGui\python\python\Lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\pytorch\auto_split.py", line 134, in pytorch_auto_split
    return auto_split(img, upscale, tiler)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\upscale\auto_split.py", line 45, in auto_split
    return split(
           ^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\upscale\auto_split.py", line 220, in _max_split
    row_result.add_tile(
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\nodes\impl\upscale\tile_blending.py", line 128, in add_tile
    self.result[:, :w, ...] = tile
    ~~~~~~~~~~~^^^^^^^^^^^^
ValueError: could not broadcast input array from shape (1936,1312) into shape (1936,1312,1)

**Comment by the-database** [comment]
This should be fixed in v1.1.6, please give it a try and let me know.

**Comment by Its-my-right** [comment]
> This should be fixed in v1.1.6, please give it a try and let me know.

I just tried 1.1.6, now there is this new error:

Upscaling with command: .\python\python\python.exe "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py" --settings C:\Users\Unknown\AppData\Roaming\MangaJaNaiConverterGui\appstate2.json
Matched Chain: {'$type': 'MangaJaNaiConverterGui.ViewModels.UpscaleChain, MangaJaNaiConverterGui', 'ChainNumber': '4', 'MinResolution': '0x0', 'MaxResolution': '0x1250', 'IsGrayscale': True, 'IsColor': False, 'MinScaleFactor': 2, 'MaxScaleFactor': 0, 'ModelFilePath': '4x_MangaJaNai_1200p_V1_ESRGAN_70k.pth', 'ModelTileSize': 'Auto (Estimate)', 'AutoAdjustLevels': True, 'ResizeHeightBeforeUpscale': 0, 'ResizeWidthBeforeUpscale': 0, 'ResizeFactorBeforeUpscale': 100.0}
Auto adjusted levels: new black level = 24; new white level = 255
save image: C:\Users\Unknown\Downloads\Mangas\Fairy Tail\1_to_dp_assemble_and_clean_blacks\Fairy Tail - T21 MANGAJANAI\Fairy Tail - T21 002-upscaled.png
Process Process-3:
Traceback (most recent call last):
  File "C:\Users\Unknown\AppData\Roaming\MangaJaNaiConverterGui\python\python\Lib\site-packages\multiprocess\process.py", line 314, in _bootstrap
    self.run()
  File "C:\Users\Unknown\AppData\Roaming\MangaJaNaiConverterGui\python\python\Lib\site-packages\multiprocess\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py", line 1288, in postprocess_worker_image
    save_image(
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py", line 647, in save_image
    image = final_target_resize(
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py", line 460, in final_target_resize
    return image_resize(
           ^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py", line 163, in image_resize
    return dotgain20_resize(image, new_size)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Local\MangaJaNaiConverterGui\current\backend\src\run_upscale.py", line 143, in dotgain20_resize
    pil_image = Image.fromarray(image, mode="L")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Unknown\AppData\Roaming\MangaJaNaiConverterGui\python\python\Lib\site-packages\PIL\Image.py", line 3113, in fromarray
    raise ValueError(msg)
ValueError: Too many dimensions: 3 > 2.
Elapsed time: 21.95 seconds

**Comment by the-database** [comment]
Thanks for testing, these should be fixed in v1.1.7, please let me know.

**Comment by Its-my-right** [comment]
> Thanks for testing, these should be fixed in v1.1.7, please let me know.

Just tested v1.1.7, this is fixed, thank you very much :)

**Comment by the-database** [comment]
Thanks for testing!
